pytest_asyncio/plugin.py

```diff
@@ -360,6 +403,10 @@ async def setup():
             res = await func(**_add_kwargs(func, kwargs, event_loop, request))
             return res
 
+        # Since the fixture doesn't have a cleanup phase, if it set any context
```

Comment:
I'm not quite sure about this part. Would it be better to copy in the variable changes and add a finalizer to reset them, as we do for generator fixtures? That would also be fairly intuitive, but the behavior of async fixtures would still be different than synchronous fixture functions.

Reply:
On further consideration, I think it's better to copy in the changes. If a generator fixture propagates contextvar changes, it seems surprising for a non-generator fixture to fail to do so.

And if changes _are_ copied in, I would rather reset on teardown than just leave them in place: creating an `asyncio.Task` usually copies the context, so it shouldn't be terribly surprising for contextvar changes to eventually go out of scope, and that reduces the risk of unexpected crosstalk between tests.